lib/plugin/captureStepAsSubtitle.js

```diff
@@ -0,0 +1,153 @@
+const { formatTimestamp } = require('subtitle');
+const { v4: uuidv4 } = require('uuid');
+const fsPromises = require('fs/promises');
+const event = require('../event');
+
+let steps = {};
+let stepCounter = 1;
+
+let testStartedAt;
+const defaultGlobalName = '__';
+/**
+ * Add descriptive nested steps for your tests:
+ *
+ * ```js
+ * Scenario('project update test', async (I) => {
+ *   __`Given`;
+ *   const projectId = await I.have('project');
+ *
+ *   __`When`;
+ *   projectPage.update(projectId, { title: 'new title' });
+ *
+ *   __`Then`;
+ *   projectPage.open(projectId);
+ *   I.see('new title', 'h1');
+ * })
+ * ```
+ * Steps prefixed with `__` will be printed as nested steps in `--steps` output:
+ *
+ * ```
+ *   Given
+ *     I have "project"
+ *   When
+ *     projectPage update
+ *   Then
+ *     projectPage open
+ *     I see "new title", "h1"
+ * ```
+ *
+ * Also those steps will be exported to allure reports.
+ *
+ * This plugin can be used
+ *
+ * ### Config
+ *
+ * * `enabled` - (default: false) enable a plugin
+ * * `registerGlobal` - (default: false) register `__` template literal function globally. You can override function global name by providing a name as a value.
+ *
+ * ### Examples
+ *
+ * Registering `__` globally:
+ *
+ * ```js
+ * plugins: {
+ *   commentStep: {
+ *     enabled: true,
+ *     registerGlobal: true
+ *   }
+ * }
+ * ```
+ *
+ * Registering `Step` globally:
+ * ```js
+ * plugins: {
+ *   commentStep: {
+ *     enabled: true,
+ *     registerGlobal: 'Step'
+ *   }
+ * }
+ * ```
+ *
+ * Using only local function names:
+ * ```js
+ * plugins: {
+ *   commentStep: {
+ *     enabled: true
+ *   }
+ * }
+ * ```
+ * Then inside a test import a comment function from a plugin.
+ * For instance, you can prepare Given/When/Then functions to use them inside tests:
+ *
+ * ```js
+ * // inside a test
+ * const step = codeceptjs.container.plugins('commentStep');
+ *
+ * const Given = () => step`Given`;
+ * const When = () => step`When`;
+ * const Then = () => step`Then`;
+ * ```
+ *
+ * Scenario('project update test', async (I) => {
+ *   Given();
+ *   const projectId = await I.have('project');
+ *
+ *   When();
+ *   projectPage.update(projectId, { title: 'new title' });
+ *
+ *   Then();
+ *   projectPage.open(projectId);
+ *   I.see('new title', 'h1');
+ * });
+ * ```
+ */
+module.exports = function (config) {
+  event.dispatcher.on(event.test.before, (test) => {
+    testStartedAt = Date.now();
+    steps = {};
```

Comment:
This code will mostly break when ran in parallel, however, as an alternative I can maintain a `key...value` pair for each test by its `id` and its corresponding steps..

But, is there any alternative for the same?

Reply:
1. No, it won't break in parallel. All tests are running in isolated workers
2. seems fine